client/src/components/requestsTable/row.tsx

```diff
@@ -9,14 +9,61 @@ import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
 import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
 import Typography from '@mui/material/Typography';
 import { Button } from '@mui/material';
-import BadgeIcon from '@mui/icons-material/Badge';
+import httpInstance from '../../services/axiosConfig';
 import './style.css';
-import { CarWithCustomerInfo } from '../../interfaces';
+import { CarsWithCustomerRow, CarWithCustomerInfo } from '../../interfaces';
+import CustomizedSnackbars from '../snackbar';
 
-function Row(props:{ car:CarWithCustomerInfo, state:string }) {
-  const { car, state } = props;
+type Props = { car:CarWithCustomerInfo, state:string,
+  setCarsData:React.Dispatch<React.SetStateAction<CarsWithCustomerRow>>,
+  setOpenSnackBar:React.Dispatch<React.SetStateAction<boolean>>,
+  setSnackbarMessage:React.Dispatch<React.SetStateAction<string>>,
+  setSnackbarType:React.Dispatch<React.SetStateAction<'success' | 'error'>>,
+};
+
+function Row(props:Props) {
+  const {
+    car, state, setCarsData, setOpenSnackBar, setSnackbarMessage, setSnackbarType,
+  } = props;
   const [open, setOpen] = useState(false);
 
+  const handleClose = (event?: React.SyntheticEvent | Event, reason?: string) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setOpenSnackBar(false);
+  };
+
+  const handleReject = async (id:number) => {
+    try {
+      setOpenSnackBar(false);
+      const response = await httpInstance.delete(`/cars/${id}`);
+      setCarsData((prevState) => prevState.filter(((element) => element.id !== car.id)));
+      setSnackbarMessage('Reject request successfully');
```

Comment:
as you can see thers's too many states for `SnackBar`, maybe we can combine them in one state as follows
```js
  const [snackbar, setSnackbar = useState({isOpen, msg, type});
```